platform/features/table/src/controllers/TelemetryTableController.js

```diff
@@ -157,33 +183,44 @@ define(
          */
         TelemetryTableController.prototype.setup = function () {
             var handle = this.handle,
+                domainObject = this.$scope.domainObject,
                 table = this.table,
-                self = this;
+                self = this,
+                metadatas = [];
 
-            if (handle) {
-                handle.promiseTelemetryObjects().then(function () {
-                    table.buildColumns(handle.getMetadata());
+            function addMetadata(object) {
+                if (object.hasCapability('telemetry') &&
+                    object.getCapability('telemetry').getMetadata()){
+                    metadatas.push(object.getCapability('telemetry').getMetadata());
+                }
+            }
 
+            function buildAndFilterColumns(){
+                if (metadatas && metadatas.length > 0){
+                    self.$scope.rows = [];
+                    table.buildColumns(metadatas);
                     self.filterColumns();
-
-                    // When table column configuration changes, (due to being
-                    // selected or deselected), filter columns appropriately.
-                    self.changeListeners.push(self.$scope.$watchCollection(
-                        'domainObject.getModel().configuration.table.columns',
-                        self.filterColumns.bind(self)
-                    ));
-                });
+                }
             }
-        };
 
-        /**
-         * @private
-         * @param object The object for which data is available (table may
-         * be composed of multiple objects)
-         * @param datum The data received from the telemetry source
-         */
-        TelemetryTableController.prototype.updateRows = function (object, datum) {
-            this.$scope.rows.push(this.table.getRowValues(object, datum));
+            //if (handle) {
```

Comment:
Style: remove code, don't comment it out.  Especially when it's resulting in unnecessary indentation.